drivers/adc/ad4170/ad4170.h

```diff
@@ -0,0 +1,1142 @@
+	/** DAC settings (registers HW_LDAC_Mask, HW_Toggle_Mask, Channel_En and DAC_Span). */
+	struct ad4170_dac_config dac;
+};
+
+/**
+ * @struct ad4170_spi_settings
+ * @brief AD4170 SPI settings.
+ */
+struct ad4170_spi_settings {
+	/** Select short instruction, 7-bit Addressing, instead of the default 15-bit Addressing. */
+	bool short_instruction;
+	/** Enable communication CRC generation and checking. */
+	bool crc_enabled;
+	/** Enable sync pattern loss detection and recovery. */
+	bool sync_loss_detect;
+};
+
+/**
+ * @enum ad4170_id
+ * @brief Device selector ID
+ */
+enum ad4170_id {
+	ID_AD4170,
+	ID_AD4171,
+	ID_AD4172,
+	ID_AD4190
+};
+
+/**
+ * @struct ad4170_init_param
+ * @brief AD4170 SPI init paraeter structure, used for initializing the ad4170_dev.
+ */
+struct ad4170_init_param {
+	/** Active Device */
+	enum ad4170_id id;
+	/** Host processor SPI configuration. */
+	struct no_os_spi_init_param spi_init;
+	/** AD4170 SPI communication configuration. */
+	struct ad4170_spi_settings spi_settings;
+	/* Rdy GPIO (EOC) monitor timeout loop counter */
+	uint32_t rdy_conv_timeout;
+	/** AD4170 device configuration. */
+	struct ad4170_config config;
+	/** SYNC_IN GPIO configuration. */
+	struct no_os_gpio_init_param *gpio_sync_inb;
+	/** DIG_AUX1 GPIO configuration. */
+	struct no_os_gpio_init_param *gpio_dig_aux1;
+	/** DIG_AUX2 GPIO configuration. */
+	struct no_os_gpio_init_param *gpio_dig_aux2;
+};
+
+/**
+ * @struct ad4170_dev
+ * @brief AD4170 device descriptor.
+ */
+struct ad4170_dev {
+	/** Device ID */
+	enum ad4170_id id;
+	/** Copy of the Host processor SPI configuration. */
+	struct no_os_spi_init_param spi_init;
+	/** SPI descriptor. */
+	struct no_os_spi_desc *spi_desc;
+	/** Copy of the AD4170 SPI communication configuration. */
+	struct ad4170_spi_settings spi_settings;
+	/* Rdy GPIO (EOC) monitor timeout loop counter */
+	uint32_t rdy_conv_timeout;
+	/** Specifies whether CPU is big endian. */
+	bool big_endian;
+	/** Copy of the AD4170 device configuration. */
+	struct ad4170_config config;
+	/** Copy of the SYNC_IN GPIO configuration. */
+	struct no_os_gpio_init_param *gpio_sync_inb_init;
+	/** SYNC_IN GPIO descriptor. */
+	struct no_os_gpio_desc *gpio_sync_inb;
+	/** Copy of the DIG_AUX1 GPIO configuration. */
+	struct no_os_gpio_init_param *gpio_dig_aux1_init;
+	/** DIG_AUX1 GPIO descriptor. */
+	struct no_os_gpio_desc *gpio_dig_aux1;
+	/** Copy of the DIG_AUX2 GPIO configuration. */
+	struct no_os_gpio_init_param *gpio_dig_aux2_init;
+	/** DIG_AUX2 may be configured as input or output, this specifies whether it is an output. */
+	bool dig_aux2_output;
+	/** DIG_AUX2 GPIO descriptor. */
+	struct no_os_gpio_desc *gpio_dig_aux2;
+	/** Specifies whether the SPI+GPIO is interface is initialized. */
+	bool digif;
+};
+
+/** ad4170_config default state when the chip is hardware/software reset. */
+extern struct ad4170_config ad4170_config_reset;
+
+int32_t ad4170_spi_reg_read(struct ad4170_dev *dev,
```

Comment:
we should stick with our convention of using `int` type for functions.

Reply:
will do